api/schemas/report.json

```diff
@@ -114,6 +114,11 @@
             ]
         },
 
+        "master": {
+            "description": "The name of the master that was used to compile the catalog. If failover occurred, this will hold the first master successfully contacted. If this run had no master (e.g. a `puppet apply` run), this field will be blank",
```

Comment:
If I understand correctly:

Question 1: when only specifying a single server and it fails, should the report `master` key report the failing server name. This question seems applicable to `server_list` as well. What if we have three potential servers and each fails? In that case, no entry seems best. Perhaps then we should do the same for a single server?

Question 2: Why are we not receiving the report on the `report_server` when using `server_list`? This will require some debugging around report submission. We want to make sure that the report server is correct. If you specify `report_server` AND `server_list`, does Puppet keep the correct value in `report_server` at the point of submission in the Configurer?